src/service/resources-impl.js

```diff
@@ -1316,6 +1343,7 @@ export class ResourcesImpl {
         // Build all resources visible, measured, and in the viewport.
         if (
           !r.isBuilt() &&
+          !r.isBuilding() &&
```

Comment:
Why did this not exist beforehand / why add it now?

Reply:
This prevents a race condition during the element's build. `isBuilt` checks the element's build, which is async and can take time to become true. `isBuilding` is checks whether the resource has started build, and is sync.

It's not noticeable, because nothing later in this pass will attempt to build elements. But if something did, then this race would cause `buildsThisPass_` to increment, even though no building would actually be done.